File: bindings/lua/Polycode/GPUDrawCall.lua
class "GPUDrawCall"


function GPUDrawCall:__getvar(name)
	if name == "options" then
		local retVal = Polycode.GPUDrawCall_get_options(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["GPUDrawOptions"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "modelMatrix" then
		local retVal = Polycode.GPUDrawCall_get_modelMatrix(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Matrix4"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function GPUDrawCall:__setvar(name,value)
	if name == "options" then
		Polycode.GPUDrawCall_set_options(self.__ptr, value.__ptr)
		return true
	elseif name == "modelMatrix" then
		Polycode.GPUDrawCall_set_modelMatrix(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function GPUDrawCall:__delete()
	if self then Polycode.delete_GPUDrawCall(self.__ptr) end
end

File: bindings/javascript/Polycode/GPUDrawCall.js
function GPUDrawCall() {
	Object.defineProperties(this, {
		'options': { enumerable: true, configurable: true, get: GPUDrawCall.prototype.__get_options, set: GPUDrawCall.prototype.__set_options},
		'modelMatrix': { enumerable: true, configurable: true, get: GPUDrawCall.prototype.__get_modelMatrix, set: GPUDrawCall.prototype.__set_modelMatrix}
	})
}
GPUDrawCall.prototype.__get_options = function() {
	var retVal = new GPUDrawOptions()
	retVal.__ptr = 	Polycode.GPUDrawCall__get_options(this.__ptr)
	return retVal
}

GPUDrawCall.prototype.__set_options = function(val) {
	Polycode.GPUDrawCall__set_options(this.__ptr, val.__ptr)
}

GPUDrawCall.prototype.__get_modelMatrix = function() {
	var retVal = new Matrix4()
	retVal.__ptr = 	Polycode.GPUDrawCall__get_modelMatrix(this.__ptr)
	return retVal
}

GPUDrawCall.prototype.__set_modelMatrix = function(val) {
	Polycode.GPUDrawCall__set_modelMatrix(this.__ptr, val.__ptr)
}

Duktape.fin(GPUDrawCall.prototype, function (x) {
	if (x === GPUDrawCall.prototype) {
		return;
	}
	Polycode.GPUDrawCall__delete(x.__ptr)
})
